app/src/main/java/com/sma2/sma2/ExerciseFragments/ExSustainedVowel.java

```diff
@@ -0,0 +1,187 @@
+        ExSustainedVowel fragment = new ExSustainedVowel();
+        return fragment;
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.fragment_ex_sustained_vowel, container, false);
+        startButton = view.findViewById(R.id.btnStartExSV);
+        startButton.setOnClickListener(this);
+        doneButton = view.findViewById(R.id.btnDoneExSV);
+        doneButton.setOnClickListener(this);
+        redoButton = view.findViewById(R.id.btnRedoExSV);
+        redoButton.setOnClickListener(this);
+        volumeBar = view.findViewById(R.id.volumeExSV);
+        recorder = SpeechRecorder.getInstance(getActivity().getApplicationContext(), new VolumeHandler(volumeBar));
+        return view;
+    }
+
+    @Override
+    public void onAttach(Context context) {
+        super.onAttach(context);
+        if (context instanceof OnFragmentInteractionListener) {
+            mListener = (OnFragmentInteractionListener) context;
+        } else {
+            throw new RuntimeException(context.toString()
+                    + " must implement OnFragmentInteractionListener");
+        }
+    }
+
+    @Override
+    public void onDetach() {
+        super.onDetach();
+        mListener = null;
+        recorder.release();
+    }
+
+    @Override
+    public void onClick(View v) {
+        int id = v.getId();
+        //Record or Stop button pressed
+        if (id == startButton.getId()){
+            //Start recording
+            if (!recording){
+                filePath = recorder.prepare("exerciseID" + "_" + "patientID");
+                recorder.record();
+                recording = true;
+                startButton.setText(R.string.stop);
+            //Stop recording
+            } else {
+                recorder.stopRecording();
+                recording = false;
+                startButton.setText(R.string.start);
+                startButton.setVisibility(View.INVISIBLE);
+                doneButton.setVisibility(View.VISIBLE);
+                redoButton.setVisibility(View.VISIBLE);
+                getActivity().runOnUiThread(new Runnable() {
+                    @Override
+                    public void run() {
+                        volumeBar.setProgress(0);
+                    }
+                });
+            }
+        //Exercise done
+        } else if (id == doneButton.getId()){
+            mListener.onExerciseFinished(filePath);
+        //Redo exercise
+        } else if (id == redoButton.getId()){
+            doneButton.setVisibility(View.INVISIBLE);
+            redoButton.setVisibility(View.INVISIBLE);
+            startButton.setVisibility(View.VISIBLE);
+            Thread deleteFile = new Thread() {
```

Comment:
This should be handled by a file handler class, as this is functionality that will be needed at multiple points. Can you convert that into an issue, meaning that we need to create such a file handler class @PKlumpp ?